ui/src/components/Inventory/InventoryFormModal.js

```diff
@@ -0,0 +1,129 @@
+import Button from '@material-ui/core/Button'
+import Dialog from '@material-ui/core/Dialog'
+import DialogActions from '@material-ui/core/DialogActions'
+import DialogContent from '@material-ui/core/DialogContent'
+import DialogTitle from '@material-ui/core/DialogTitle'
+import Grid from '@material-ui/core/Grid'
+import React from 'react'
+import TextField from '../Form/TextField'
+import { Field, Form, Formik } from 'formik'
+
+class InventoryFormModal extends React.Component {
+  render() {
+      const {
+        formName,
+        handleDialog,
+        handleInventory,
+        title,
+        initialValues,
+        units
+      } = this.props
+
+      return (
+        <Dialog
+        open={this.props.isDialogOpen}
+        maxWidth='sm'
+        fullWidth={true}
+        onClose={() => { handleDialog(false) }}
+        >
+          <Formik
+            initialValues={initialValues}
+            onSubmit={values => {
+              console.log('values after submit', values)
+              values.bestBeforeDate = values.bestBeforeDate + 'T12:00:00Z'
+              handleInventory(values)
+              handleDialog(true)
+            }}>
+            {helpers =>
+              <Form
+              noValidate
```

Comment:
We want form validation to ensure required fields are being set and with the correct values